nixos/modules/services/monitoring/rustdesk-server.nix

```
@@ -0,0 +1,96 @@
+{ lib, pkgs, config, ... }:
+{
+  options.services.rustdesk-server = with lib; with types; {
+    enable = mkEnableOption "enable RustDesk";
+
+    package = mkOption {
+      type = package;
+      description = "The directus package to use";
+      default = pkgs.rustdesk-server;
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open the connection ports.
+        TCP (21115, 21116, 21117, 21118, 21119)
+        UDP (21116)
+      '';
+    };
+
+    relayIP = mkOption {
+      type = str;
+      description = ''
+        The RustDesk relay has to know its public facing IP
+      '';
+    };
+  };
+
+  config = let
+    cfg = config.services.rustdesk-server;
+    serviceDefaults = {
+      enable = true;
+      requiredBy = [ "rustdesk.target" ];
+      serviceConfig = {
+        Slice = "rustdesk.slice";
+        User  = "rustdesk";
+        Group = "rustdesk";
+        Environment = [];
+        WorkingDirectory = "/var/lib/rustdesk";
+        StateDirectory   = "rustdesk";
+        StateDirectoryMode = "0750";
+        LockPersonality = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
+        PrivateTmp = true;
+        PrivateUsers = true;
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
+        ProtectSystem = "strict";
+        RemoveIPC = true;
+        RestrictNamespaces = true;
+        RestrictSUIDSGID = true;
+      };
+    };
+  in lib.mkIf cfg.enable {
+    users.users.rustdesk = {
+      description = "System user for RustDesk";
+      isSystemUser = true;
+      group = "rustdesk";
+    };
+    users.groups.rustdesk = {};
+
+    networking.firewall.allowedTCPPorts = [ 21115 21116 21117 21118 21119 ];
+    networking.firewall.allowedUDPPorts = [ 21116 ];
```

Maybe these were meant to be guarded by `cfg.openFirewall`?

```suggestion
    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ 21115 21116 21117 21118 21119 ];
    networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [ 21116 ];
```